dts/bindings/clock-mgmt/clock-device.yaml

```diff
@@ -0,0 +1,55 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  This file needs to be included by devices that need to specify clock
+  controller states. The maximum number of states currently defined is 5
+  (clock-state-0...clock-state-4) but can be incremented if required.
+  Clock state nodes are used to configure a clock controller, by setting
+  properties on clock nodes declared as children of the clock controller.
+  Clock states may represent a full clock tree, but there is no requirement
+  for them to.
+properties:
+  clock-outputs:
```

Comment:
Yes, that is why. I want to keep the `clocks` property backwards compatible because the clock control framework needs to be able to coexist with the clock management framework while we transition to the new framework.

Ideally I'd like to make it possible for downstream users to disable the clock management framework simply via Kconfig if they aren't ready to migrate their board to use it (since doing so will require defining clock management states for peripherals they are using)

If we do reuse the `clocks` property, a user who wishes to remain on the clock control framework and keep clock management disabled would also need to rewrite the `clocks` property for any peripheral they were using to restore it to the value previously used.

Reply:
eventually, to align with other DT projects like linux, it would be better to call the property `clocks`, would we plan to do a mass rename of this later ? or will clock control framework exist indefinitely?

Reply:
I think a mass rename would make the most sense, yeah. The long term goal would be to deprecate the clock control framework. At that point we could (and should) do a mass rename of the property.